pkg/controller/account/account_controller.go

```diff
@@ -313,16 +316,67 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 
 			// Create access key and role for BYOC account
 			if currentAcctInstance.Status.State == "" {
+
+				// retrive current accessKeyID for comparison later
+				initAccessKeyID, err := r.getAWSAccessKeyId(accountClaim)
+				if err != nil {
+					reqLogger.Error(err, "Failed to retrieve accountClaim accessKeyID")
+					return reconcile.Result{}, err
+				}
+
 				// Rotate access keys
-				err = r.byocRotateAccessKeys(reqLogger, byocAWSClient, accountClaim)
+				err = r.byocRotateAccessKeys(reqLogger, awsSetupClient, accountClaim)
 				if err != nil {
 					reqLogger.Info("Failed to rotate BYOC access keys")
 					r.accountClaimBYOCError(reqLogger, accountClaim, err)
 					return reconcile.Result{}, err
 				}
 
+				// Create new client after rotating access keys
+				// Retrieve now-rotated key
+				rotatedAccessKeyID, err := r.getAWSAccessKeyId(accountClaim)
+				if err != nil {
+					reqLogger.Error(err, "Failed to retrieve rotated accountClaim accessKeyID")
+					return reconcile.Result{}, err
+				}
+
+				// validate secret has changed
+				reqLogger.Info("Validating rotated keys have been updated")
+				for i := 1; i <= 20; i++ {
+					// keys have updated
+					if initAccessKeyID != rotatedAccessKeyID {
+						break
+					} else {
+						// reque request if keys match
+						time.Sleep(500 * time.Millisecond)
+					}
+					// This may need tuning.
+					if i == 20 {
+						reqLogger.Info("Timeout attempting to validate rotated keys")
+					}
+				}
+				reqLogger.Info("Rotated keys successfully validated")
```

Comment:
Should this logic go into a separate function and but put inside `byocRotateAccessKeys()` ?

Reply:
It could. I would like to submit as is and make a note to refactor this and every other occasion of key rotation. Compare secrets/ validate client as a utility.